compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1358,6 +1358,34 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                             Applicability::MaybeIncorrect,
                         );
                     } else {
+                        let is_mut = mut_ref_self_ty_satisfies_pred || ref_inner_ty_mut;
+                        let sugg_prefix = format!("&{}", if is_mut { "mut " } else { "" });
+                        let sugg_msg = &format!(
+                            "consider{} borrowing here",
+                            if is_mut { " mutably" } else { "" }
+                        );
+
+                        // Issue #109436, we need to add parentheses properly for method calls
+                        // for example, `foo.into()` should be `(&foo).into()`
+                        if let Ok(snippet) = self
+                            .tcx
+                            .sess
+                            .source_map()
+                            .span_to_snippet(self.tcx.sess.source_map().next_point(span))
```

Comment:
Would this detect something like this?

```rust
foo
    .into()
```

It looks like `span_lookahead` might be a better option than `next_point`.

Reply:
Got it! Thanks!